### ISSUE TITLE
False positive with var-declaration

### ISSUE THREAD

**Issue by rliebz** [mgechev/revive#1152]
**Describe the bug**
The `var-declaration` check incorrectly warns if explicitly assigning a zero value in a `var` declaration, even if the zero value being assigned isn't the default for a variable of that type.

**To Reproduce**
1. I updated revive `go install github.com/mgechev/revive@latest`
2. I run it with the following flags:

```shell
revive main.go
```

main.go:

```go
// Package main is a package.
package main

func main() {
	var a any = 0
	_ = a

	var b any = ""
	_ = b
}
```

**Expected behavior**
I would expect no errors for `var-declaration`, because the zero value for a variable of type `any` is `nil`, not `0` or `""`.

**Logs**
```
main.go:5:14: should drop = 0 from declaration of var a; it is the zero value
main.go:8:14: should drop = "" from declaration of var b; it is the zero value
```

**Desktop (please complete the following information):**
 - OS: macOS 15.1.1
 - Version of Go: `go version go1.23.3 darwin/arm64`

**Additional context**
N/A

**Comment by chavacava** [comment]
Hi @rliebz thanks for reporting the problem. I'll take a look

**Comment by denisvmedia** [comment]
@chavacava can you please check my PR?

**Comment by chavacava** [comment]
> @chavacava can you please check my PR?

Yes, no problem nut please in the future assign you the issue before start working on it.

**Comment by denisvmedia** [comment]
@chavacava yeah, sorry, missed that.

**Comment by denisvmedia** [comment]
@rliebz please check with my updates merged.

**Comment by rliebz** [comment]
Confirmed fixed, thanks for the quick response time!
